src/engine/PathSearch.cpp

```diff
@@ -473,8 +473,17 @@ void PathSearch::pathsToResultTable(IdTable& tableDyn, PathsLimited& paths,
 }
 
 // _____________________________________________________________________________
-std::unique_ptr<Operation> PathSearch::clone() const {
+std::unique_ptr<Operation> PathSearch::cloneImpl() const {
   auto copy = std::make_unique<PathSearch>(*this);
   copy->subtree_ = subtree_->clone();
+  if (sourceTree_.has_value()) {
+    copy->sourceTree_ = sourceTree_.value()->clone();
+  }
+  if (targetTree_.has_value()) {
+    copy->targetTree_ = targetTree_.value()->clone();
+  }
+  if (sourceAndTargetTree_.has_value()) {
+    copy->sourceAndTargetTree_ = sourceAndTargetTree_.value()->clone();
+  }
```

Comment:
Code duplication, can be something like
`cloneIfNonEmpty(copy->sourceTree_);`
(same for the other two).